File: 3753-maximum-difference-between-even-and-odd-frequency-i/3753-maximum-difference-between-even-and-odd-frequency-i.java
class Solution {
    public int maxDifference(String s) {
       int n=s.length();
       Map<Character,Integer> m=new HashMap<>();
       for(int i=0;i<n;i++){
            char ch=s.charAt(i);
            m.put(ch,m.getOrDefault(ch,0)+1);
       }   

       int e=100,o=Integer.MIN_VALUE;
       for(int fre:m.values()){
            if(fre%2==0){
                if(fre<e) e=fre;
            }
            else{
                if(fre>o) o=fre;
            }
       }
    //    System.out.print(e);
    //    System.out.print(o);
    // if (e == Integer.MIN_VALUE) e=0;
    // if (o== Integer.MIN_VALUE) o=0;
       return o-e;
    }
}

File: 3753-maximum-difference-between-even-and-odd-frequency-i/3753-maximum-difference-between-even-and-odd-frequency-i.cpp
class Solution {
public:
    int maxDifference(string s) {
        int n=s.size();
        sort(s.begin(),s.end());
        int e=100,o=1,c=1;
        for(int i=1;i<n;i++){
            if(s[i]==s[i-1]){
                c++;
            }
            else{
                if(c%2==0){
                    if(c<e) e=c;
                }
                else{
                    if(c>o) o=c;
                }
                c=1;
            }
        }
        if(c%2==0){
                    if(c<e) e=c;
                }
                else{
                    if(c>o) o=c;
                }
        return o-e;
    }
};